src/sentry/sentry_metrics/querying/data_v2/execution.py

```diff
@@ -560,13 +618,23 @@ def schedule(
 
         Note that this method won't execute the query, since it's lazy in nature.
         """
-        executable_query = ExecutableQuery(
-            with_series=True,
-            with_totals=True,
-            # We identify the query with its index.
-            identifier=str(len(self._scheduled_queries)),
+        # For now, we are always building a (totals -> series) query, but the execution engine is fully capable of
+        # supporting either a single totals or series query.
+        executable_query = ScheduledQuery(
+            type=ScheduledQueryType.TOTALS,
             metrics_query=query,
+            next=ScheduledQuery(
+                type=ScheduledQueryType.SERIES,
+                metrics_query=query,
+                order=order,
+                limit=limit,
+            ),
             order=order,
             limit=limit,
         )
+        # We initialize the query by performing type-aware mutations that prepare the query to be executed correctly
+        # (e.g., adding `totals` to a totals query...).
+        executable_query = executable_query.initialize(
```

Comment:
This is the step where queries are mutated based on their type. Doing that here makes it easier to write the executor.